### PR TITLE
Implement loading source

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: c-layer/lib.cxx [hunk 1]
```diff
@@ -272,4 +275,53 @@ struct RepositoryList list_repositories(struct Status *status) noexcept {
...
+  }
+  try {
+    zypp::RepoInfo zypp_repo = repo_manager->getRepo(alias);
+    if (zypp_repo == zypp::RepoInfo::noRepo) {
+      status->state = status->STATE_FAILED;
+      status->error = format_alloc("Cannot load repo with alias %s. Repo not found.", alias);
+      return;
+    }
+
+    // NOTE: progress parameter is ignored, so do not make sense to pass progress
```
Comment (mvidner) [inline c-layer/lib.cxx:293]:
```suggestion
    // NOTE: loadFromCache has an optional `progress` parameter but it ignores it anyway
```

**Comment by mvidner** [comment]
@jreidinger as you introduced an `int enabled;` field into the API, it made me replace all boolean `int`s with real `bool`s :)

**Review by mvidner — APPROVED** [review]

File: rust/zypp-agama-sys/src/bindings.rs [hunk 2]
```diff
@@ -1,5 +1,8 @@
 /* automatically generated by rust-bindgen 0.70.1 */
 
+pub const __bool_true_false_are_defined: u32 = 1;
+pub const true_: u32 = 1;
+pub const false_: u32 = 0;
```
Comment (jreidinger) [inline rust/zypp-agama-sys/src/bindings.rs:5]: looks like stdbool use identical definition as me :)

File: rust/zypp-agama/src/lib.rs [hunk 3]
```diff
@@ -27,17 +28,13 @@ pub struct Repository {
...
     F: FnMut(i64, String) -> bool,
 {
     let user_data = &mut *(user_data as *mut F);
     let res = user_data(zypp_data.value, string_from_ptr(zypp_data.name));
-    if res {
-        1 as c_int
-    } else {
-        0 as c_int
-    }
+    res
```
Comment (jreidinger) [inline rust/zypp-agama/src/lib.rs:37]: here it can be direct result of user_data call..but lets do it in another PR
